### PR TITLE
New App ingresse

### PR DESCRIPTION
**Pull request by gondekr**
- Events
- Home
- Login
- SignUp

### CODE REVIEW

File: IngresseSDK/Model/Venue.swift [hunk 1]
```diff
@@ -19,4 +19,21 @@ public class Venue: JSONConvertible {
...
+    public override func applyJSON(_ json: [String : Any]) {
+        for (key,value) in json {
+            if key == "location" {
+                guard
+                    let location = value as? [Double],
+                    location.count >= 2
+                    else { continue }
+
+                applyKey("latitude", value: location[0])
+                applyKey("latitude", value: location[1])
```
Comment (bissuh) [inline IngresseSDK/Model/Venue.swift]: Maybe it should be longitude in 31 and latitude in 32.

File: IngresseSDK/Model/Planner.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//
+//  Planner.swift
+//  IngresseSDK
+//
+//  Created by Rubens Gondek on 12/5/17.
+//  Copyright © 2017 Ingresse. All rights reserved.
+//
+
+public class Planner: JSONConvertible {
```
Comment (bissuh) [inline IngresseSDK/Model/Planner.swift]: Must add @objcMembers for swift 4.

File: IngresseSDK/Model/IngresseUser.swift [hunk 3]
```diff
@@ -9,7 +9,7 @@
 public class IngresseUser: JSONConvertible {
```
Comment (bissuh) [inline IngresseSDK/Model/IngresseUser.swift]: I would ass @objcMembers for swift 4

File: IngresseSDK/Model/Highlight.swift [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//
+//  Highlight.swift
+//  IngresseSDK
+//
+//  Created by Rubens Gondek on 12/5/17.
+//  Copyright © 2017 Ingresse. All rights reserved.
+//
+
+public class Highlight: JSONConvertible {
```
Comment (bissuh) [inline IngresseSDK/Model/Highlight.swift]: I would ass @objcmembers for swift 4

File: IngresseSDK/Model/EventDate.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
+//
+//  EventDate.swift
+//  IngresseSDK
+//
+//  Created by Rubens Gondek on 12/5/17.
+//  Copyright © 2017 Ingresse. All rights reserved.
+//
+
+public class EventDate: JSONConvertible {
```
Comment (bissuh) [inline IngresseSDK/Model/EventDate.swift]: I would ass @objcmembers for swift 4

File: IngresseSDK/Parser/JSONConvertible.swift [hunk 6]
```diff
@@ -32,16 +32,21 @@ public class JSONConvertible: NSObject {
...
 
-        guard
-            self.responds(to: NSSelectorFromString(key)),
-        let dictionary = value as? [String:Any],
-        let data = dictionary["data"] as? [[String:Any]]
-        else { return }
+        guard self.responds(to: NSSelectorFromString(key)) else { return }
+
+        var data: [[String:Any]]?
+        if let dictionary = value as? [String:Any] {
```
Comment (bissuh) [inline IngresseSDK/Parser/JSONConvertible.swift]: Remove else from if statement.

File: IngresseSDK/Services/EventService.swift [hunk 7]
```diff
@@ -55,4 +62,91 @@ public class EventService: BaseService {
...
+    ///   - searchTerm: Term to filter event list
+    ///   - delegate: Callback listener
+    public func getEvents(page: Int,
+        from place: String,
+        ofCategory category: Int,
+        andSearchTerm searchTerm: String,
+        delegate: EventSyncDelegate) {
+        let path = categories[category]
+
+        let fields = "id,title,planner,date,venue,rsvp,saleEnabled,type,link,poster,description"
```
Comment (bissuh) [inline IngresseSDK/Services/EventService.swift]: Not mandatory, but I would add the fields as an additional, not required, parameter with the fields below as default.

Please, see feature/owls branch as reference. I believe we have a method like this out there.

File: IngresseSDK/Services/UserService.swift [hunk 8]
```diff
@@ -35,5 +35,80 @@ public class UserService: BaseService {
...
+
+        let firstName = splitName.joined(separator: " ")
+        params["name"] = firstName
+
+        params["phone"] = phone
+        params["email"] = email
+        params["emailConfirm"] = email
+        params["password"] = password
+        params["passCheck"] = password
+        params["terms"] = true
```
Comment (bissuh) [inline IngresseSDK/Services/UserService.swift]: Is there a reason so this is a hard coded information?

File: IngresseSDK/Services/EventService.swift (hunk 7 above)
Comment (gondekr) [reply IngresseSDK/Services/EventService.swift]: the parameter "fields" is not required to call this function, as you can see on its header, these fields are requested by default

File: IngresseSDK/Model/Planner.swift (hunk 2 above)
Comment (gondekr) [reply IngresseSDK/Model/Planner.swift]: It's added on super class, JSONConvertible is marked with @objcmembers already and all of its subclasses conforms to it as well

**Pull request merged**